### PR TITLE
[BUGFIX]fix python 3.8 ctypes dll load with windows

### PR DESCRIPTION
**Pull request by yajiedesign**
## Description ##
fix python 3.8 ctypes dll load with windows

### Essentials ###
- [X] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [X] Changes are complete (i.e. I finished coding on this PR)

### CODE REVIEW

**Comment by yajiedesign** [comment]
@szha @aaronmarkham Lint doesn't know the new parameter. What should I do?

**Comment by yajiedesign** [comment]
I disable it.

File: python/mxnet/base.py [hunk 1]
```diff
@@ -276,7 +276,15 @@ class MXCallbackList(ctypes.Structure):
...
-    lib = ctypes.CDLL(lib_path[0], ctypes.RTLD_LOCAL)
+    if sys.version_info >= (3, 8):
+        if os.name == "nt":
+            # use LOAD_WITH_ALTERED_SEARCH_PATH, For simplicity, let's just fill the numbers.
+            # pylint: disable=E1123
+            lib = ctypes.CDLL(lib_path[0], winmode=0x00000008)
+        else:
+            lib = ctypes.CDLL(lib_path[0], ctypes.RTLD_LOCAL)
+    else:
+        lib = ctypes.CDLL(lib_path[0], ctypes.RTLD_LOCAL)
```
Comment (szha) [inline python/mxnet/base.py:287]: nit:
```suggestion
    if sys.version_info >= (3, 8) and os.name == "nt":
        # use LOAD_WITH_ALTERED_SEARCH_PATH, For simplicity, let's just fill the numbers.
        # pylint: disable=E1123
        lib = ctypes.CDLL(lib_path[0], winmode=0x00000008)
    else:
        lib = ctypes.CDLL(lib_path[0], ctypes.RTLD_LOCAL)
```

**Comment by yajiedesign** [comment]
run ci [windows-gpu]

**Comment by wkcn** [comment]
https://docs.python.org/3/library/ctypes.html

The document shows that 

> The winmode parameter is used on Windows to specify how the library is loaded (since mode is ignored). It takes any value that is valid for the Win32 API LoadLibraryEx flags parameter. When omitted, the default is to use the flags that result in the most secure DLL load to avoiding issues such as DLL hijacking. Passing the full path to the DLL is the safest way to ensure the correct library and dependencies are loaded.

Could we pass the full path of DLL by ` lib = ctypes.CDLL(os.path.abspath(lib_path[0]))` ?

**Comment by yajiedesign** [comment]
@wkcn The key is to find CUDA's dll. Adjusting mxnet.dll path doesn't help.

File: python/mxnet/base.py [hunk 2]
```diff
@@ -276,7 +276,12 @@ class MXCallbackList(ctypes.Structure):
 def _load_lib():
     """Load library by searching possible path."""
     lib_path = libinfo.find_lib_path()
-    lib = ctypes.CDLL(lib_path[0], ctypes.RTLD_LOCAL)
+    if sys.version_info >= (3, 8) and os.name == "nt":
+        # use LOAD_WITH_ALTERED_SEARCH_PATH, For simplicity, let's just fill the numbers.
+        # pylint: disable=E1123
+        lib = ctypes.CDLL(lib_path[0], winmode=0x00000008)
```
Comment (marcoabreu) [inline python/mxnet/base.py:282]: Is the magic value available as a constant somewhere?

Comment (yajiedesign) [reply python/mxnet/base.py:282]: Not at the moment.

**Pull request merged**